Glyssen/BookScript.cs

```diff
@@ -102,36 +103,50 @@ public System.Collections.Generic.IReadOnlyList<Block> GetScriptBlocks(bool join
 			if (!join || m_blockCount == 0)
 				return m_blocks;
 
-			var list = new List<Block>(m_blockCount);
-			list.Add(m_blocks[0]);
-			for (int i = 1; i < m_blockCount; i++)
+			var list = new List<Block>(m_blockCount) {m_blocks[0]};
+
+			for (var i = 1; i < m_blockCount; i++)
 			{
 				var block = m_blocks[i];
+				var prevBlock = list.Last();
+
 				if (!block.IsParagraphStart)
 				{
-					var prevBlock = list.Last();
-					if (block.CharacterIdInScript == prevBlock.CharacterIdInScript && (block.Delivery ?? string.Empty) == (prevBlock.Delivery ?? String.Empty))
+					if (block.CharacterIdInScript == prevBlock.CharacterIdInScript && (block.Delivery ?? string.Empty) == (prevBlock.Delivery ?? string.Empty))
 					{
-						var newBlock = prevBlock.Clone();
-						int skip = 0;
-						if (prevBlock.BlockElements.Last() is ScriptText && block.BlockElements.First() is ScriptText)
-						{
-							var lastScriptText = (ScriptText)newBlock.BlockElements.Last();
-							lastScriptText.Content += ((ScriptText)block.BlockElements.First()).Content;
-							skip = 1;
-						}
-						foreach (var blockElement in block.BlockElements.Skip(skip))
-							newBlock.BlockElements.Add(blockElement.Clone());
-						newBlock.UserConfirmed &= block.UserConfirmed;
-						list[list.Count - 1] = newBlock;
+						list[list.Count - 1] = CombineBlockWithPreviousBlock(block, prevBlock);
 						continue;
 					}
 				}
+				else if (block.StyleTag.StartsWith("q")
```

Comment:
Seems like it would be "better" to access the IsPoetic flag from the "TextProperties" of the style from the USFM stylesheet, rather than relying on the convention of starting these markers with the letter "q". Not sure how easily we can fluff up a stylsheet in this context (see USXParser for example).